.github/workflows/deploy-docker.yml

```diff
@@ -0,0 +1,57 @@
+# Builds the Docker image and publishes to the GitHub container registry
+
+name: Publish Docker Image
+
+on:
+  push:
+    branches: ["main"]
+    # Publish semver tags as releases.
+    tags: ["v*.*.*"]
+
+env:
+  # TODO: Also publish to Codeberg
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Log in to container registry
+        # docker/login-action@v3.3.0
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567
```

Comment:
Why the SHA instead of the normal version?

Reply:
Paranoia about the publish flow

Reply:
The odds are incredibly slim that a malicious actor would re-tag the docker actions to a malicious commit, but pinning to the hash avoids that entirely